roomacoustics/sti.py

```diff
@@ -0,0 +1,176 @@
+noise and masking aspects.
+
+STI considers 7 octaves between 125 Hz and 8 kHz (125 Hz is not considered for
+the female STI) and 14 modulation frequencies between 0.63 Hz and 12 Hz. 
+
+
+============================
+
+Reference
+---------
+..  [1] IEC 60268-16:2011
+    Sound system equipment - Part 16: Objective rating of speech intelligibility 
+    by speech transmission index
+
+"""
+
+import warnings
+from pyfar import Signal  
+import pyfar.dsp.filter as filt
+import numpy as np
+
+def sti(signal, data_type=None, gender='male', level=np.array([np.nan]), 
+            snr=np.array([np.nan])):
+    """
+    Parameters
+    ---------
+    signal : Signal
+        The IR to be analyzed. Length must be at least 1.6 s and not shorter than
+        1/2 RT60. [1],  p. ??
+
+    data_type : 'electrical', 'acoustical'
+        Determines weather input signals are obtained acoustically or electrically.
+        Auditory effects can only be considered when "acoustical" [1], p. ??. 
+        Default is 'none'.
+
+    gender: 'female', 'male'
+        Defines the applied weighting factors. Default is 'male' as the STI 
+        algorithm is initially based on the male speech spectrum.
+
+    noise: np.array, 0
+        Level of the test signal in octave for bands 125 Hz - 8000 Hz in dB_SPL.
+
+    snr: np.array, 0
+        SNR in octave for bands 125 Hz - 8000 Hz in dB.
+
+    """
+    
+
+# process signal, check input
+    sig = signal.copy()
+    #2D Signal
+    sig = sig.flatten()
+
+#check for sufficient signal length (IEC 60268-16 p. ## and p. ##) 
+    if signal.n_samples/sig.sampling_rate < 1.6:
+        warnings.warn("Signal length below 1.6 seconds.")
+
+# apply octave band filters (preliminary with crossover; later: perf. 
+# reconstructing oct. filter)
+    sig_oct = (filt.crossover(sig, 4, [90, 180, 355, 710, 1400, 2800, 5600,
+    11200])[1:-1,:])
+# MTF per octave and modulation frequency
+    mf = [0.63, 0.80, 1, 1.25, 1.60, 2, 2.5, 3.15, 4, 5, 6.3, 8, 10, 12]
+    mtf = np.zeros((14,)+sig_oct.cshape)
+    sig_en = np.sum(np.square(sig_oct.time),axis=-1)
+    sig_oct_newlength = sig_oct.copy()
+
+    #calculate mtf
+    with np.errstate(divide='ignore'):  #return nan for empty IR
+        for i in range(len(mf)):
+            fbin = int(np.ceil(mf[i]*sig_oct.n_samples/sig_oct.sampling_rate))
+            newlength = int(sig_oct.sampling_rate*fbin/mf[i])
+            sig_oct_newlength.time = np.concatenate((sig_oct.time, 
+                np.zeros((sig_oct.cshape + (newlength-sig_oct.n_samples,)))),
+                axis=-1)
+            sig_oct_newlength_sq = Signal(np.square(sig_oct_newlength.time),
+                sig_oct.sampling_rate)
+            mtf[i]=abs(sig_oct_newlength_sq.freq[...,fbin])/sig_en
+
+
+# Adjustment of mtf for ambient noise, auditory masking and threshold effects
+
+    # calculation only if levels are given
+    if not np.isnan(level).all():
```

Comment:
- According to Section 6.1 in the DIN, SNR and level effects can be applied separately. If I'm not mistaken, I would suggest to make this two different section/if-cases

I would suggest the following here

```
if snr is not none:
    snr = np.asarray(snr).flatten()  # make it a flat array in case the user past a list (which would be fine)
    if snr.size != correctSize:  # raise a ValueError instead of a warning if the number of entries is wrong
        raise ValueError("Some message")
```
The check for the level could be done in the same way :)